### PR TITLE
VAGOV-TEAM-90731: add ombinfo prop

### PR DESCRIPTION
**Pull request by derekhouck**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- Extracts the normalized `ombInfo` section created by department-of-veterans-affairs/content-build#2252 from fetched form data and inserts it into the `IntroductionPage` component.

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#90731

## Testing done

- New unit tests for the `ombInfo` action and reducer.
- New unit tests for the app's new root reducer.
- New unit tests for the `IntroductionPage` container.
- Updated tests for the `form-load` reducer and the `App` container.
- Local testing using mock data.

## Screenshots

Local host rendering a mock form with dynamic OMB Info data:
Screenshot 2024-09-05 at 2 26 23 PM

## What areas of the site does it impact?

This PR only touches the `form-renderer` application, which is not yet live.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Due to our use of [`createRoutesWithSaveInProgress`](https://github.com/department-of-veterans-affairs/vets-website/blob/4a1a414377f3c2f4a6a7db449a28706de8cebcf2/src/applications/form-renderer/routes.jsx#L7) to create our child routes, I could not easily pass along data outside of `formConfig` to the `IntroductionPage` component as props. Instead, I loaded it into our Redux store and pulled from that. This is relevant to future work, as we will eventually want to customize both `IntroductionPage` and `ConfirmationPage` more thoroughly on a per-form basis. Neither of these components required access to the Redux store before this PR, so continuing on in this direction will make both components more complex.

### CODE REVIEW

File: src/applications/form-renderer/tests/unit/containers/App.unit.spec.js [hunk 1]
```diff
@@ -51,7 +51,8 @@ describe('<App /> component', () => {
 
 describe('<App /> integration', () => {
   const rootReducer = combineReducers({
-    formLoad: formLoadReducer.formLoad,
+    formLoad: reducer.formLoad,
+    ombInfo: reducer.ombInfo,
```
Comment (ryguyk) [inline src/applications/form-renderer/tests/unit/containers/App.unit.spec.js:55]: This feels odd to me. See more general discussion.

TL;DR it seems like loading OMB info should be _part of_ loading the other form data.

File: src/applications/form-renderer/actions/form-load/index.js [hunk 2]
```diff
@@ -61,6 +62,7 @@ export const fetchFormConfig = (
       const form = findFormByFormId(forms, formId);
       const formConfig = createFormConfig(form);
 
+      dispatch(ombInfoLoaded(form.ombInfo));
       dispatch(formLoadingSucceeded(formConfig));
```
Comment (ryguyk) [inline src/applications/form-renderer/actions/form-load/index.js:66]: Again, see more general discussion. It feels a bit odd to have the OMB data treated separately from the rest of the form data.

**Review by ryguyk — COMMENTED** [review]
First off, beautiful code. This is really well written, organized, etc.

I wonder, though, if this code is implementing the cleanest approach. It feels a bit odd to me to treat the loading (and storage via Redux) of the OMB info in a manner that is isolated from the rest of the form data that is loaded and subsequently converted into a `formConfig` object. 

This code snippet, specifically, feels like it should already be doing what we want to do:

```
// src/applications/form-renderer/utils/formConfig.js (before)

const formConfig = createFormConfig(form);

dispatch(formLoadingSucceeded(formConfig));
```

It feels odd to have to _also_ call a `dispatch` for OMB-specific info:
```
// src/applications/form-renderer/utils/formConfig.js (proposed change)

const formConfig = createFormConfig(form);

dispatch(ombInfoLoaded(form.ombInfo));
dispatch(formLoadingSucceeded(formConfig));
```

It'd be great if the `formConfig` object would already encapsulate the needed OMB info so that we wouldn't need to introduce it separately on the Redux store. I think we can do that:

```
// src/applications/form-renderer/utils/formConfig.js (createFormConfig function)
return {
    ...
    introduction: props => (
      <IntroductionPage {...props} ombInfo={ombInfo} />
    ),
    ...
}
```

If we need to abstract this behavior for reuse, we could create a higher-order component:

```
const withProps = (WrappedComponent, additionalProps) => {
    return (props) => <WrappedComponent {...props} {...additionalProps} />;
};
```

I'm certainly open to other thoughts on this. Perhaps we can discuss.

**Comment by derekhouck** [comment]
> ```
> // src/applications/form-renderer/utils/formConfig.js (createFormConfig function)
> return {
>     ...
>     introduction: props => (
>       <IntroductionPage {...props} ombInfo={ombInfo} />
>     ),
>     ...
> }
> ```

- Yes, this would be a lot simpler and cleaner. Never even occurred to me. After I finish the revisions on department-of-veterans-affairs/content-build#2252, I'll give this a stab and verify the props get successfully passed through using this method. I appreciate the insight.

**Review by ryguyk — DISMISSED** [review]
Nice work!

File: src/applications/form-renderer/containers/IntroductionPage.jsx [hunk 3]
```diff
@@ -1,100 +1,119 @@
...
+        messages={formConfig.savedFormMessages}
+        pageList={pageList}
+        startText="Start the Application"
+      />
+      <p />
+      <va-omb-info
+        data-testid="va-omb-info"
+        exp-date={ombInfo.expDate}
+        omb-number={ombInfo.ombNumber}
+        res-burden={ombInfo.resBurden}
```
[the remark this answers is not in the document]
Comment (derekhouck) [reply src/applications/form-renderer/containers/IntroductionPage.jsx]: I agree it _shouldn't_ be undefined, but in case it is for some reason, I've updated `IntroductionPage` to omit the `va-omb-info` component when `ombInfo` is falsy. My reasoning here is that the form should still render and be functional even if there was a mistake at the VA that led to this information not being included. CAIA may have a different point-of-view on this, but we can tackle that later should it become an issue.

File: src/applications/form-renderer/utils/formConfig.js [hunk 4]
```diff
@@ -68,7 +69,9 @@ export const createFormConfig = form => {
     trackingPrefix: `${formId}-`,
     // eslint-disable-next-line no-console
     submit: () => console.log(`Submitted ${subTitle}`),
-    introduction: IntroductionPage,
+    introduction: props => (
+      <IntroductionPage {...props} ombInfo={form.ombInfo} />
```
[the remark this answers is not in the document]
Comment (derekhouck) [reply src/applications/form-renderer/utils/formConfig.js]: I made your suggest change for where I'm grabbing `ombInfo` on the page. See my comment in `IntroductionPage` for how I'm handling when `ombInfo` is undefined.

**Review by ryguyk — APPROVED** [review]
